openwebwork/webwork2
# Develop rename update

This adds the ability to change the courseTitle and the courseInstitution in addition to changing the courseID for the course. Changing the courseID involves renaming several tables in the database, changing the title or the institution simply changes a field in the database. title and institution were set when a course was created and recorded in the database but their was no way to change their values.

To test:  

There are four items to be chosen in order to rename the database.  oldCourseID (from a scrolling menu)
newCourseID, newCourseTitle and newCourseInstitution.  Each of the last three has both a checkbox to indicate that it is to be changed and an input field for the new value. 

You always need to select an old courseID but there are many combinations of checkbox and input field that need to be checked.  If the checkbox is checked and the corresponding input field is empty there should be an error message (after submit).  If the input field has data but the checkbox is not checked then no operation should occur.  If the checkbox is not checked and the input field is empty there should not be an error.  Finally if the checkbox is checked and the input field is not empty then the report should indicate that the title or institution or ID has been changed. There are 64 combinations to check.  

After the initial submit there should be an accurate report of what errors need to be corrected (e.g. a field needs to be filled in) and what changes will be made. 

Cancel buttons should take you back to the first rename page with the checkboxs and input fields in their original state.

If you have direct access to the database make sure that reported changes are actually made.
